packages/dart_frog_cli/lib/src/dev_server_runner/dev_server_runner.dart

```diff
@@ -236,13 +315,60 @@ class DevServerRunner {
     }
 
     final watcher = _directoryWatcher(path.join(cwdPath));
-    final subscription = watcher.events
+    _watcherSubscription = watcher.events
         .where(shouldReload)
         .debounce(Duration.zero)
         .listen((_) => _reload());
 
-    await subscription.asFuture<void>();
-    await subscription.cancel();
-    return ExitCode.success;
+    _watcherSubscription!.asFuture<void>().then((_) async {
+      await _cancelWatcherSubscription();
+      await stop();
```

Comment:
This is an example of prioritizing the local operation, now regarding the cancellation of the subscription